superset/utils/core.py

```diff
@@ -126,9 +126,13 @@ def __repr__(self):
         """Return the function's docstring."""
         return self.func.__doc__
 
+    def __copy__(self):
+        return _memoized(self.func, self.watch)
+
     def __get__(self, obj, objtype):
         if not self.is_method:
-            self.is_method = True
+            copy = self.__copy__()
+            return functools.partial(copy.__call__, obj)
```

Comment:
Could you provide some more context as to what you’re trying to achieve here?

Reply:
The analyzer at lgtm.com finds this assignment harmful (https://lgtm.com/rules/910074/). 

The problem is (AFAIU) that we are mutating a descriptor object, and this can cause surprises to different instances of the same class. The recommendation is to create a copy that contains the desired state and return that instead.

In a patch I haven't pushed yet, I do a `copy.is_method = True` before returning as I do here. I wanted to copy the present instance, mutate it, and return it.

This, however, makes the unit tests `test_memoized_on_methods_with_watches` and `test_memoized_on_methods` fail.

Reply:
On further thought I think that the code uses this pattern to actually achieve the desired result. What I propose, in this case, is to either silence the alert with `# lgtm[py/mutable-descriptor]`, or to just ignore it. Maybe a comment in the code would be enough?

Reply:
@john-bodley kindly when you find a minute can you advise on this?